Lib/ufo2ft/filters/optimizeAnchors.py

```diff
@@ -17,6 +17,14 @@ def filter(self, glyph):
             # We're a base!
             return False
 
+        # More sanity checks: skip over spacing marks
+        if glyph.width != 0:
+            return False
+        # Also skip over marks which are deliberately positioned over the
+        # previous glyphs
+        if glyph.getBounds().xMax < 0:
```

Comment:
as a heuristic as to guess whether a mark was "deliberately positioned over the previous glyph", I think this is a bit too fragile. It's conceivable that the xMax still be >= 0 while the mark being intentionally designed to fall on top of another (e.g. narrow) base glyph